src/main/java/com/YOGIITSU/service/SignUpService.java

```diff
@@ -1,88 +1,80 @@
+        // 2. 이메일 인증 확인
+        Optional<EmailMessage> verifiedEmail = emailMessageRepository.findByEmailAndIsApprovedTrue(
+            dto.getEmail());
+        if (verifiedEmail.isEmpty()) {
+            throw new IllegalArgumentException("이메일 인증이 완료되지 않았습니다.");
         }
 
-        // 이메일 도메인 검사
-        if (!memberSignUpRequestDto.getEmail().endsWith("@suwon.ac.kr")) {
-            response.put("message", "이메일은 무조건 @suwon.ac.kr로 끝나야 합니다.");
-            return response;
+        // 3. 아이디 중복 체크
+        if (memberRepository.findByMemberId(dto.getMemberId()).isPresent()) {
+            throw new IllegalArgumentException("아이디가 이미 존재합니다.");
         }
 
-        // 3. 이름 중복 체크
-        if (memberRepository.findByUserName(memberSignUpRequestDto.getUserName()).isPresent()) {
-            response.put("message", "해당 이름으로 가입한 내역이 있습니다.");
-            return response;
+        // 4. 이메일 도메인 검사
+        if (!dto.getEmail().endsWith("@suwon.ac.kr")) {
+            throw new IllegalArgumentException("이메일은 무조건 @suwon.ac.kr로 끝나야 합니다.");
         }
 
-        // 4. 이름 길이 검사 (최소 2글자 이상)
-        if (memberSignUpRequestDto.getUserName().length() < 2) {
-            response.put("message", "이름은 최소 2글자 이상이어야 합니다.");
-            return response;
+        // 5. 이름 중복 체크
+        if (memberRepository.findByUserName(dto.getUserName()).isPresent()) {
+            throw new IllegalArgumentException("해당 이름으로 가입한 내역이 있습니다.");
         }
 
-        // 5. 비밀번호 유효성 검사
-        try {
-            validatePassword(memberSignUpRequestDto.getPassword());
-        } catch (IllegalArgumentException e) {
-            response.put("message", e.getMessage());
-            return response;
+        // 6. 이름 길이 검사
+        if (dto.getUserName().length() < 2) {
+            throw new IllegalArgumentException("이름은 최소 2글자 이상이어야 합니다.");
         }
 
-        // 6. 비밀번호 암호화
-        String encodedPassword = passwordEncoder.encode(memberSignUpRequestDto.getPassword());
+        // 7. 비밀번호 유효성 검사
+        validatePassword(dto.getPassword());
+
+        // 8. 비밀번호 암호화 및 회원 생성
+        String encodedPassword = passwordEncoder.encode(dto.getPassword());
 
-        // 7. Member 객체 생성
         Member member = Member.builder()
-            .memberId(memberSignUpRequestDto.getMemberId())
+            .memberId(dto.getMemberId())
             .password(encodedPassword)
-            .email(memberSignUpRequestDto.getEmail())
-            .userName(memberSignUpRequestDto.getUserName())
-            .role("USER")
+            .email(dto.getEmail())
+            .userName(dto.getUserName())
+            .role("USER")  // 반드시 ROLE_ 접두사
             .joinAt(java.time.LocalDateTime.now())
             .build();
 
-        // 8. 데이터 저장
+        // 9. DB 저장
         memberRepository.save(member);
-
-        // 9. 성공 응답 반환
-        response.put("message", "회원가입이 성공적으로 완료되었습니다.");
-        return response;
     }
 
-    // 비밀번호 유효성 검사 메서드
     private void validatePassword(String password) {
```

Comment:
지금은 `"^[a-z0-9!@#...]"` 이렇게 돼 있어서 대문자 들어가면 실패하거나,  
소문자가 없는데도 통과될 수 있어서 조건이 모호한데

 **"문자 + 숫자 + 특수문자 + 8자 이상"** 식으로 각각 따로 체크해주면 좋을 듯!
```suggestion
    private void validatePassword(String password) {
    if (password.length() < 8) {
        throw new IllegalArgumentException("비밀번호는 최소 8자 이상이어야 합니다.");
    }
    if (!Pattern.compile("[a-zA-Z]").matcher(password).find()) {
        throw new IllegalArgumentException("비밀번호에는 영문자가 최소 1자 이상 포함되어야 합니다.");
    }
    if (!Pattern.compile("[0-9]").matcher(password).find()) {
        throw new IllegalArgumentException("비밀번호에는 숫자가 최소 1자 이상 포함되어야 합니다.");
    }
    if (!Pattern.compile("[!@#$%^&*(),.?\":{}|<>]").matcher(password).find()) {
        throw new IllegalArgumentException("비밀번호에는 특수문자가 최소 1자 이상 포함되어야 합니다.");
    }
}
```

Reply:
아ㅏ 오키!